plugin/pkg/replacer/replacer_test.go

```diff
@@ -247,6 +247,10 @@ func TestLabels(t *testing.T) {
 
 	for lbl := range labels {
 		repl := replacer.Replace(ctx, state, w, lbl)
+		// test empty state and/or recorder won't panic
+		replacer.Replace(ctx, request.Request{}, w, lbl)
+		replacer.Replace(ctx, state, nil, lbl)
```

Comment:
I mean, you need to keep the original call
```go
  repl := replacer.Replace(ctx, state, w, lbl)
```
and the second call with empty Request and Recorder
```go
  repl := replacer.Replace(ctx, request.Request{}, nil, lbl)
```
Four calls looks like overkill. Anyway, depending on `lbl` it will check either Request{} or Recorder.